stable_baselines3/common/policies.py

```diff
@@ -620,9 +620,12 @@ def optimizer(self) -> th.optim.Optimizer:
         try:
             return self._optimizer
         except AttributeError:
-            self._optimizer = self.optimizer_class(self.parameters(), lr=self.lr_schedule(1), **self.optimizer_kwargs)
+            self._optimizer = self._make_optimizer()
         return self._optimizer
 
+    def _make_optimizer(self) -> th.optim.Optimizer:
```

Comment:
Defer optimizer creation so it can be done with the parameters in GPU. This allows us to use a fused implementation of Adam.

Reply:
I don't understand how is the optimizer creation deferred with this line change. Wouldn't the optimizer still be created at the same point in code execution? Or do we call `_make_optimizer` again after the parameters are loaded on the GPU?

Reply:
Yeah, makes sense!